lib/curl_setup.h

```diff
@@ -277,10 +277,23 @@
 #  include <exec/execbase.h>
 #  include <proto/exec.h>
 #  include <proto/dos.h>
+#  include <unistd.h>
+/*
+ * pwd.h required for getpwuid but works only if __NO_NET_API is not
+ * defined so it must be defined here.
+ */
+#  ifdef HAVE_PWD_H
+#    include <pwd.h>
+#  endif
```

Comment:
Shouldn't the `pwd.h` fix rather be in the only two files that actually use this header? so that it doesn't have to be included unconditionally in every single source code.

Reply:
I have chosen this place as here are Amiga specific includes and definitions. Probably (likely) I can move this to the relevant places by writing

```
#ifdef HAVE_PWD_H
#undef  __NO_NET_API
#include <pwd.h>
#endif
```